lib/assignDisabledRanges.mjs

```diff
@@ -10,23 +10,27 @@ import {
 	isConfigurationComment,
 } from './utils/configurationComment.mjs';
 import { assert, assertNumber, assertString } from './utils/validateTypes.mjs';
+import { isAtRule, isComment, isDeclaration, isRule } from './utils/typeGuards.mjs';
 import isStandardSyntaxComment from './utils/isStandardSyntaxComment.mjs';
 
-/** @import {Comment as PostcssComment, Document as PostcssDocument, Root as PostcssRoot} from 'postcss' */
+import { isTokenComment, tokenize } from '@csstools/css-tokenizer';
+
+/** @import {Node as PostcssNode, Comment as PostcssComment, Document as PostcssDocument, Root as PostcssRoot, Position as PostcssPosition} from 'postcss' */
 /** @import {DisabledRange, DisabledRangeObject, PostcssResult} from 'stylelint' */
+/** @typedef {{start?: PostcssPosition, end?: PostcssPosition}} Source */
```

Comment:
[suggest] Minor. Using the `Source` type may be a bit clearer. But you can keep it as-is.

```suggestion
/** @import {Node as PostcssNode, Comment as PostcssComment, Document as PostcssDocument, Root as PostcssRoot, Source as PostcssSource} from 'postcss' */
/** @import {DisabledRange, DisabledRangeObject, PostcssResult} from 'stylelint' */
/** @typedef {Pick<PostcssSource, 'start' | 'end'>} Source */
```